test/auth/auth_test.go

```diff
@@ -277,3 +277,29 @@ func getAccessToken(t *testing.T, weaviateUrl, user, pw string) (string, string)
 	require.Nil(t, err)
 	return tokens.AccessToken, tokens.RefreshToken
 }
+
+func TestApiKey(t *testing.T) {
+	url := fmt.Sprintf("127.0.0.1:%v", testsuit.WCSPort)
+	headers := map[string]string{}
+	cfg := weaviate.Config{Host: url, Scheme: "http", Headers: headers}
+	var err error
+	cfg, err = weaviate.AddAuthClient(cfg, auth.ApiKeys{ApiKey: "my-secret-key"}, 60*time.Second)
+	assert.Nil(t, err)
+
+	client := weaviate.New(cfg)
+	AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+	assert.Nil(t, AuthErr)
+}
+
+func TestWrongApiKey(t *testing.T) {
+	url := fmt.Sprintf("127.0.0.1:%v", testsuit.WCSPort)
+	headers := map[string]string{}
+	cfg := weaviate.Config{Host: url, Scheme: "http", Headers: headers}
+	var err error
+	cfg, err = weaviate.AddAuthClient(cfg, auth.ApiKeys{ApiKey: "wrong_key"}, 60*time.Second)
+	assert.Nil(t, err)
+
+	client := weaviate.New(cfg)
+	AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+	assert.NotNil(t, AuthErr)
```

Comment:
both of theses test should be there for `weaviate.AddAuthClient` and `NewConfig`